### PR TITLE
[#166792020] New service list section

### PR DESCRIPTION
**Pull request by valentinacala**
This pr introduces the new service list design as an experimental feature.

NewServiceSection

To solve the bolder section header style on Android, it can be changed the **H5 component default font-weight from 600 to 300** (while fontweight sets by screen styles is ignored)

image

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#166792020](https://www.pivotaltracker.com/story/show/166792020): Come CIT voglio vedere il list item dei servizi con il nuovo layout

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/preferences/OldServicesHomeScreen.tsx [hunk 1]
```diff
@@ -143,19 +147,23 @@ export const getAllSections = createSelector(
     const orgfiscalCodes = Object.keys(services.byOrgFiscalCode);
     return orgfiscalCodes
       .map(fiscalCode => {
-        const title = organizations[fiscalCode] || fiscalCode;
```
Comment (Undermaken) [inline ts/screens/preferences/OldServicesHomeScreen.tsx]: you could remove **organizationsSelector**  and use [organizationNamesByFiscalCodeSelector](https://github.com/teamdigitale/io-app/blob/d5ed3c0b5beebe81b31794f05cdc2e3622715319/ts/store/reducers/entities/organizations/organizationsByFiscalCodeReducer.ts)

Comment (valentinacala) [reply ts/screens/preferences/OldServicesHomeScreen.tsx]: updated within https://github.com/teamdigitale/io-app/pull/1214/commits/79d09b50b604bdbd71b7e0e7db0433977eadca36

File: ts/screens/preferences/OldServicesHomeScreen.tsx [hunk 2]
```diff
@@ -143,19 +147,23 @@ export const getAllSections = createSelector(
...
-          title,
+          organizationName,
+          organizationFiscalCode,
           data
         };
       })
       .filter(_ => _.data.length > 0)
       .sort((a, b) =>
-        (a.title || "").localeCompare(b.title, "it", { caseFirst: "lower" })
+        (a.organizationName || "").localeCompare(b.organizationName, "it", {
```
[the remark this answers is not in the document]
Comment (valentinacala) [reply ts/screens/preferences/OldServicesHomeScreen.tsx]: Actually, I just updated the attribute name. Shouldn't that change being applied by the merged 1217 pr?

File: ts/components/services/NewServiceListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  listItem: {
+    marginLeft: customVariables.contentPadding,
+    marginRight: customVariables.contentPadding
+  }
+});
+
+export default class NewServiceListItem extends React.PureComponent<Props> {
+  public render() {
+    const potService = this.props.item;
+    const onPress = pot.getOrElse(
```
Comment (Undermaken) [inline ts/components/services/NewServiceListItem.tsx]: render function is called at every change of state
Can we avoid to create a function each time?
I'm thinking about a solution like this, what do you think about it?

```javascript
private handleItemPressed(potService: pot.Pot<ServicePublic, Error>): void {
    if (pot.isSome(potService)) {
      this.props.onSelect(potService.value);
    }
  }
```

Comment (valentinacala) [reply ts/components/services/NewServiceListItem.tsx]: Examining the component, it seems we cannot avoid calling an arrow function when declaring onPress: the function we have to pass to the component wants the selected service as input while the onPress attribute wants a function with no inputs. For the case you mentioned, it need to be:
 `onPress={() => this.handleItemPressed(potService)}` 

I will look to search if any other solution exist to overcome it.

File: ts/components/screens/ListItemComponent.tsx [hunk 4]
```diff
@@ -1,21 +1,23 @@
...
 import H5 from "../ui/H5";
 import IconFont from "./../ui/IconFont";
 import { BadgeComponent } from "./BadgeComponent";
 
 type Props = Readonly<{
   title: string;
-  subTitle: string;
-  onPress: () => void;
+  onPress?: () => void;
+  subTitle?: string;
```
[the remark this answers is not in the document]
Comment (valentinacala) [reply ts/components/screens/ListItemComponent.tsx]: you're right. Updated within https://github.com/teamdigitale/io-app/pull/1214/commits/8741c32d8f76554cf005655eec48dbec25b4d600

File: ts/components/services/NewServiceListItem.tsx (hunk 3 above)
Comment (Undermaken) [reply ts/components/services/NewServiceListItem.tsx]: Yes you are right!
arrow function returns a new function every time
So if don't find any better solution, leave the code as it is

File: ts/screens/preferences/OldServicesHomeScreen.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Undermaken) [reply ts/screens/preferences/OldServicesHomeScreen.tsx]: Yes it will

**Pull request merged**